### ISSUE TITLE
[Table] Resizing columns via double-click should resize each column appropriately

### ISSUE THREAD

**Issue by cmslewis** [palantir/blueprint#1115]
### Bug report

- __Package version(s)__: 1.17.1
- __Browser and OS versions__: Chrome (58) on OS X (10.11.6)

#### Steps to reproduce

1. Drag-select a range of columns.
1. Double-click on various column-resize handles within the selection.

#### Actual behavior

Columns resize to the minimum required width of the _double-clicked column._ If you double-click different columns in the selection, the column widths continually change. This isn't particularly helpful.

![2017-05-17 16 50 52](https://cloud.githubusercontent.com/assets/443450/26180956/6cd53cf0-3b21-11e7-9bc8-d0f66c7edf0d.gif)

#### Expected behavior

Columns resize to _their own respective_ minimum required widths. If you double-click different columns in the selection, nothing needs to change, because each column is already as wide as it needs to be (and no wider).

**Comment by giladgray** [comment]
wait wait wait, i thought we _designed_ it to be this way, definitely not a bug in the system: this behavior is consistent with other spreadsheet applications.

**Comment by cmslewis** [comment]
@giladgray this isn't actually consistent with other spreadsheet applications.

See Google Sheets:
![2017-05-18 10 35 14](https://cloud.githubusercontent.com/assets/443450/26215588/c36fd61c-3bb5-11e7-8e83-8b9386e161a0.gif)

See Excel:
![2017-05-18 10 37 03](https://cloud.githubusercontent.com/assets/443450/26215634/fd691ed2-3bb5-11e7-9f74-f9b39409ec33.gif)

**Comment by llorca** [comment]
I don't believe we designed this part. We designed the drag to resize part which works well, but we haven't touched the double-click to resize part.

Feature request or enhancement sounds good, this will be a great UX improvement 👍

**Comment by giladgray** [comment]
_oh snap look at those gifs!_ i was dead sure they did what we do but damn. carry on then.
